platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -145,9 +144,10 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
         }
         _snapshotCallback = NULL;
     });
-    dispatch_async(queue, ^{
-        _mbglMapSnapshotter->snapshot(_snapshotCallback->self());
 
+    dispatch_async(queue, ^{
+        __typeof__(self) strongSelf = weakSelf;
```

Comment:
Also return early here before dereferencing `strongSelf`.

Reply:
I think I may be misunderstanding: this doesn’t explicitly deference anything right now — should it?

Reply:
The line below dereferences `strongSelf`. It’s important to check for a `nil` `strongSelf`, not a `nil` `weakSelf`.

Reply:
Ah, alright — this is done in https://github.com/mapbox/mapbox-gl-native/pull/11771/commits/fc8be4becd1b0c4f609b35f058200e0c2b1d4d64. Chatted with @julianrex a bit about this (and, separately, the potentially problematic block/ivar interactions in this snapshot code) and he agreed with your suggestion. Thanks to you both for walking me through this. 🙇